ServerCore/Helpers/MailHelper.cs

```diff
@@ -123,22 +124,23 @@ public void SendPlaintextWithoutBcc(IEnumerable<string> recipients, string subje
     /// </summary>
     public void SendHtmlWithoutBcc(IEnumerable<string> recipients, string subject, string body)
     {
-        SendInternal(recipients, false, subject, body, true);
+        SendInternalAsync(recipients, false, subject, body, true);
     }
 
-    private void SendInternal(IEnumerable<string> recipients, bool bcc, string subject, string body, bool isHtml)
+    private async void SendInternalAsync(IEnumerable<string> recipients, bool bcc, string subject, string body, bool isHtml)
     {
         List<string> addresses = FlattenAddressLists(recipients);
         while (addresses.Count > 0)
         {
             int n = Math.Min(addresses.Count, MAX_RECIPIENTS);
             List<string> someAddresses = addresses.GetRange(0, n);
             addresses.RemoveRange(0, n);
-            SendInternal2(someAddresses, bcc, subject, body, isHtml);
+            SendInternal2Async(someAddresses, bcc, subject, body, isHtml);
```

Comment:
We should await this too

Reply:
I didn't do that because
(1) when I add the await there I get an error "Cannot await 'void'" and I couldn't figure out how to fix that.
(2) There's an await inside SendInternal2Async at the bottom where the real work is (the api call) and the code up to that point is fairly simple so it seemed like that stacked await was just extra overhead. If I think of await as a promise, there's not much reason to create a promise of a promise.

Reply:
Oh right - the pattern here is to make SendInternal2Async be a Task instead of a void. I missed that part. Right now, SendInternal2Async is returning when the SendMailAsync call begins, not when it ends. If you make it return a Task then it will return the object that can be awaited to know when the send is complete.